### PR TITLE
Upgrade dev build tooling (Babel 6 -> 7, Webpack 1 -> 4)

### PR DESCRIPTION
**Pull request by bvaughn**
Our build tooling was super outdated, which made maintenance kind of a chore. I think this step, while not totally necessary, will help with issues like #1214 and #1090.

## Build sizes
I also decided (at least unless someone knows of a good reason _not_ to do this) to stick with Webpack's default minification behavior– which is to say, do it for production builds. This resulted in the following built size reductions:

#### Standalone:
* backend.js: 215KB -> 97KB
* standalone.js: 1.3MB -> 601KB

#### Browser extension:
* packed and zipped extension: 1.8MB -> 840KB
* backend.js: 395KB -> 152KB
* panel.js: 1.3MB -> 606KB

## Build times
I timed `yarn build:standalone` and `yarn build:extension:chrome` in master and this branch and didn't notice any real difference in durations. (The duration varies between runs but seems to be about the same overall.)

## Testing
I've tested the example app, plain shell, Firefox and Chrome extensions, standalone DevTools, and the theme viewer. All seem okay after this change.

I also updated the backend/integration Webpack config, although I'm pretty sure that test hasn't been used in a long time– since the source has an invalid require (referencing a file that was moved a while back).

### CODE REVIEW

File: test/example/target.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ class Todos extends React.Component {
     todos.some(item => {
       if (item.id === id) {
         item.completed = completed;
-        text = item.text;
+        text = item.title;
```
Comment (bvaughn) [inline test/example/target.js]: This happened to be broken before. I noticed while testing the new config.

File: backend/integration/webpack.config.js [hunk 2]
```diff
@@ -29,23 +32,14 @@ module.exports = {
...
-      test: /\.js$/,
-      loader:  'babel',
-      exclude: /node_modules/,
-    }],
+    rules: [
+      {
+        test: /\.js$/,
+        loader: 'babel-loader',
+        options: {
+          ...JSON.parse(readFileSync(resolve(__dirname, '../../.babelrc'))),
```
Comment (hugmanrique) [inline backend/integration/webpack.config.js]: Is this spread operator really needed? You could remove the object braces and set `options` to the result of JSON.parse

Maybe I'm missing something

File: packages/react-devtools-core/webpack.backend.js [hunk 3]
```diff
@@ -9,38 +9,37 @@
...
-      test: /\.js$/,
-      loader: 'babel',
-      exclude: /node_modules/,
-    }],
+    rules: [
+      {
+        test: /\.js$/,
+        loader: 'babel-loader',
+        options: {
+          ...JSON.parse(readFileSync(resolve(__dirname, '../../.babelrc'))),
```
Comment (hugmanrique) [inline packages/react-devtools-core/webpack.backend.js]: Same thing as above

File: backend/integration/webpack.config.js (hunk 2 above)
Comment (bvaughn) [reply backend/integration/webpack.config.js]: Originally, I thought I might add some custom options to individual configs but that turned out not to be necessary.

**Comment by bvaughn** [comment]
I am going to land this. I'll take responsibility for anything that pops up as a result and fix it with a follow-up PR. I think the more interesting PR to get feedback on is #1236 which this soft blocks.

**Pull request merged**